logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb

```diff
@@ -2,13 +2,20 @@
 
 module LogStash; module Util
   class WrappedSynchronousQueue
+    java_import java.util.concurrent.ArrayBlockingQueue
     java_import java.util.concurrent.SynchronousQueue
     java_import java.util.concurrent.TimeUnit
 
-    def initialize
-      @queue = java.util.concurrent.SynchronousQueue.new
+    def initialize (size = 0)
+      if size == 0
```

Comment:
@jordansissel wops this needs to go away, the `drainTo` logic won't work with a `SynchronousQueue` anyways ... sorry leftover from some experiments, will delete